### PR TITLE
rename clientside components, add client for server communication

### PR DESCRIPTION
**Pull request by eelcovdw**
## Description
NOTE: sync plugin contains some changes from #397 . Merging this PR before #397 will make a small conflict in how we validate permissions in `sync_action.py`, for the rest they are compatible

Renames:
- client2.py -> core.py
- SyftClient -> SyftBoxRunner
    - Note: we dont use client here, because then the naming of actual clients is confusing.
- Plugins -> PluginManagerInterface
- SyftClientInterface -> SyftBoxContextInterface
- In plugins: var `client` is now called `context`

New:
- Move server API endpoints to `SyftBoxClient(ClientBase)`
- add to `LocalSyftBox.client: SyftBoxClient` and `Context.client: SyftBoxClient`
- different APIs are added to different namespaces in SyftBoxClient:

```python
class SyncClient(ClientBase):
    def download(self, path: Path) -> bytes: ...

class SyftBoxClient(ClientBase):
    def __init__(self, conn: httpx.Client):
        self.conn = conn
        self.sync = SyncClient(conn)

# usage
client = SyftBoxClient.from_config(my_syftbox_config)
client.sync.download(Path("my_file"))
```

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: syftbox/app/cli.py [hunk 1]
```diff
@@ -128,10 +133,10 @@ def env(with_syftbox: bool = False):
 #     pass
 
 
-def get_client(config_path: Path) -> SyftClientInterface:
+def get_syftbox_context(config_path: Path) -> SyftBoxContextInterface:
     try:
         conf = SyftClientConfig.load(config_path)
```
Comment (koenvanderveen) [inline syftbox/app/cli.py:138]: this is still called client

File: syftbox/client/core.py [hunk 2]
```diff
@@ -332,11 +283,11 @@ def run_migration(config: SyftClientConfig, migrate_datasite=True):
             shutil.rmtree(str(old_sync_state.parent))
 
 
-def run_client(
+def run_syftbox(
```
Comment (koenvanderveen) [inline syftbox/client/core.py:286]: maybe somehow indicate that this is the client side? or do you think that is clear enough from the path?

File: syftbox/client/plugins/sync/consumer.py [hunk 3]
```diff
@@ -6,43 +6,42 @@
...
 from syftbox.client.plugins.sync.sync_action import SyncAction, determine_sync_action
-from syftbox.client.plugins.sync.sync_client import SyncClient
 from syftbox.client.plugins.sync.types import SyncActionType
 from syftbox.lib.hash import hash_file
 from syftbox.lib.ignore import filter_ignored_paths
 from syftbox.server.models.sync_models import FileMetadata
 
 
-def create_local_batch(sync_client: SyncClient, paths_to_download: list[Path]) -> list[str]:
+def create_local_batch(context: SyftBoxContextInterface, paths_to_download: list[Path]) -> list[str]:
```
Comment (koenvanderveen) [inline syftbox/client/plugins/sync/consumer.py:26]: I would perhaps pass in a client here and not a context

**Comment by eelcovdw** [comment]
> I think in general there are two things that I have bigger questions about.
> 
> 1. why does context have a client?
> 2. wouldnt it be better to pass a LocalSyftbox instance around instead of a context

Discussed this in call, recap:

- The logic between Context and SyftBoxRunner is a bit double, causing some code duplication. Renamed the main syftbox class to SyftBoxRunner to make this clearer, but we might want to revisit the structures and interfaces in the future
- Context.client is just there for convenience, plugins could also manage their own client (in plugin: `SyftBoxClient.from_context(ctx)`). Leaving it as-is for now, can make this change in the future if we like it better

**Review by koenvanderveen — APPROVED** [review]

**Pull request merged**